pkg/validation/app.go

```diff
@@ -97,6 +103,58 @@ func (v *Validator) validateConfig(ctx context.Context, cr v1alpha1.App) error {
 	return nil
 }
 
+func (v *Validator) validateMetadataConstraint(ctx context.Context, cr v1alpha1.App) error {
+	name := key.AppCatalogEntryName(key.CatalogName(cr), key.AppName(cr), key.Version(cr))
+
+	entry, err := v.g8sClient.ApplicationV1alpha1().AppCatalogEntries(metav1.NamespaceDefault).Get(ctx, name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		v.logger.Debugf(ctx, "appCatalogEntry %#q not found, skipping validation", name)
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if entry.Spec.Restrictions == nil {
+		// no-op
+		return nil
+	}
+
+	if entry.Spec.Restrictions.FixedNamespace != "" {
+		if entry.Spec.Restrictions.FixedNamespace != cr.Spec.Namespace {
+			return microerror.Maskf(validationError, "app %#q could be installed in namespace %#q only, not %#q",
+				cr.Spec.Name, entry.Spec.Restrictions.FixedNamespace, cr.Spec.Namespace)
+		}
+	}
+
+	var apps *v1alpha1.AppList
+	if entry.Spec.Restrictions.ClusterSingleton || entry.Spec.Restrictions.NamespaceSingleton {
+		lo := metav1.ListOptions{
+			FieldSelector: fmt.Sprintf("metadata.name!=%s", cr.Name),
```

Comment:
Avoid comparing the same objects.

Reply:
I'm not sure comparing metadata.name is correct. The user could have installed the same app just with a different name.

I think we should compare with .spec.name or even the app.kubernetes.io/name label. The label could help if the customer has installed their own version of the app.

WDYT?

Reply:
Hi, This is to avoid comparing the same app CRs. It will not filter any apps with the same app but only the same app CRs.

Reply:
Ah got it. Not enough ☕ 😅 

That's a neat way to exclude the current CR. 👍